### PR TITLE
[HTTP] Max Wait Time Configuration Option for Connection Throttling in Client Connector

### PR DESCRIPTION
**Pull request by irunika**
## Purpose
At the moment we can only configure the number of connections in client connector for connection throttling. But we need a way to configure the maximum wait time for a given connection to wait.

This PR resolves https://github.com/ballerina-lang/ballerina/issues/4689

## Goals
Make the connection throttling wait time configurable.

## Approach
Creare a new struct for connection throttling in Options struct and add a new variable to retrieve maximum connection wait time.

## Automation tests
 - Unit tests 
   > No
 - Integration tests
   > No

## Security checks
 - Followed secure coding standards in http://wso2.com/technical-reports/wso2-secure-engineering-guidelines? yes
 - Ran FindSecurityBugs plugin and verified report? no
 - Confirmed that this PR doesn't commit any keys, passwords, tokens, usernames, or other secrets? yes

## Samples
```ballerina
endpoint<http:HttpClient> httpEndpoint {
  create http:HttpClient("http://localhost:9090/hello", {connectionThrottling: {maxActiveConnections: 1, waitTime: 2000}});
}
```

### CODE REVIEW

File: stdlib/ballerina-http/src/main/ballerina/ballerina/net/http/natives.bal [hunk 1]
```diff
@@ -337,6 +336,12 @@ public struct Options {
     SSL ssl;
     Retry retryConfig;
     Proxy proxy;
+    ConnectionThrottling connectionThrottling;
+}
+
+public struct ConnectionThrottling {
```
Comment (pubudu91) [inline stdlib/ballerina-http/src/main/ballerina/ballerina/net/http/natives.bal]: Need to document this struct

File: stdlib/ballerina-http/src/main/java/org/ballerinalang/net/http/HttpConstants.java [hunk 2]
```diff
@@ -228,6 +228,11 @@
     public static final int PROXY_USER_NAME_INDEX = 1;
     public static final int PROXY_PASSWORD_INDEX = 2;
 
+    //Connection Throttling struct indexes
+    public static final int CONNECTION_THROTTLING_STRUCT_INDEX = 4;
+    public static final int MAX_ACTIVE_CONNECTIONS_INDEX = 0;
```
Comment (chamil321) [inline stdlib/ballerina-http/src/main/java/org/ballerinalang/net/http/HttpConstants.java]: Shall we rename variable as "CONNECTION_THROTTLING_MAX_ACTIVE_CONNECTIONS_INDEX"  or  "THROTTLING_MAX_ACTIVE_CONNECTIONS_INDEX" to relate the struct?

File: stdlib/ballerina-http/src/main/java/org/ballerinalang/net/http/actions/Init.java [hunk 3]
```diff
@@ -102,11 +102,26 @@ public ConnectorFuture execute(Context context) {
...
+                if (!isInteger(maxActiveConnections)) {
+                    throw new BallerinaConnectorException("invalid maxActiveConnections value: "
+                                                                  + maxActiveConnections);
+                }
+                properties.put(HttpConstants.MAX_ACTIVE_CONNECTIONS_PER_POOL, (int) maxActiveConnections);
+
+                long waitTime = connectionThrottling.getIntField(HttpConstants.WAIT_TIME_INDEX);
+                if (!isInteger(waitTime)) {
+                    throw new BallerinaConnectorException("invalid waitTime value: "
+                                                                  + waitTime);
```
Comment (chamil321) [inline stdlib/ballerina-http/src/main/java/org/ballerinalang/net/http/actions/Init.java]: move this content to above line.

File: stdlib/ballerina-http/src/main/ballerina/ballerina/net/http/natives.bal [hunk 4]
```diff
@@ -337,6 +335,15 @@ public struct Options {
     SSL ssl;
     Retry retryConfig;
     Proxy proxy;
+    ConnectionThrottling connectionThrottling;
+}
+
+@Description { value:"This struct represents the options to be used for connection throttling" }
+@Field {value:"maxActiveConnections: No of maximum active connections for connection throttling. Default value is -1, indicated the no of connections are not restricted"}
```
Comment (pubudu91) [inline stdlib/ballerina-http/src/main/ballerina/ballerina/net/http/natives.bal]: Shall we either make this `No.` or `Number`

File: stdlib/ballerina-http/src/main/ballerina/ballerina/net/http/natives.bal [hunk 5]
```diff
@@ -337,6 +335,15 @@ public struct Options {
...
+@Field {value:"waitTime: Maximum waiting time for request to grab a idle connection from the client connector (default value: 6000 milliseconds)"}
```
Comment (pubudu91) [inline stdlib/ballerina-http/src/main/ballerina/ballerina/net/http/natives.bal]: `an idle connection`

**Pull request merged**